management_api_app/main.py

```diff
@@ -27,10 +29,30 @@ def get_application() -> FastAPI:
     return application
 
 
+def initialize_logging(logging_level: int):
+    """
+    Adds the Application Insights handler for the root logger and sets the given logging level.
+
+    :param logging_level: The logging level to set e.g., logging.WARNING.
+    """
+    logger = logging.getLogger()
+    app_insights_instrumentation_key = os.getenv("APPINSIGHTS_INSTRUMENTATIONKEY")
+
+    try:
+        logger.addHandler(AzureLogHandler(connection_string=f"InstrumentationKey={app_insights_instrumentation_key}"))
+    except ValueError:
+        logger.error("Application Insights instrumentation key missing")
+
+    logging.basicConfig(level=logging_level)
+    logger.setLevel(logging_level)
+
+
 if DEBUG:
```

Comment:
I know this is not part of the PR - but... since you're here - there is also a config.DEBUG that we could use instead of a random DEBUG flag, if you would be so kind and use that here.